client/express-checkout/utils/checkPaymentMethodIsAvailable.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import ReactDOM from 'react-dom';
+import { ExpressCheckoutElement, Elements } from '@stripe/react-stripe-js';
+import { memoize } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { isLinkEnabled } from 'wcpay/checkout/utils/upe';
+import request from 'wcpay/checkout/utils/request';
+import WCPayAPI from 'wcpay/checkout/api';
+import { getUPEConfig } from 'wcpay/utils/checkout';
+
+export const checkPaymentMethodIsAvailable = memoize(
+	( paymentMethod, cart, resolve ) => {
+		const root = ReactDOM.createRoot(
+			document.getElementById(
+				`express-checkout-check-availability-container-${ paymentMethod }`
+			)
+		);
+
+		const api = new WCPayAPI(
+			{
+				publishableKey: getUPEConfig( 'publishableKey' ),
+				accountId: getUPEConfig( 'accountId' ),
+				forceNetworkSavedCards: getUPEConfig(
+					'forceNetworkSavedCards'
+				),
+				locale: getUPEConfig( 'locale' ),
+				isStripeLinkEnabled: isLinkEnabled(
+					getUPEConfig( 'paymentMethodsConfig' )
+				),
+			},
+			request
+		);
+
+		root.render(
+			<Elements
+				stripe={ api.loadStripe() }
+				options={ {
+					mode: 'payment',
+					paymentMethodCreation: 'manual',
+					amount: Number( cart.cartTotals.total_price ),
+					currency: cart.cartTotals.currency_code.toLowerCase(),
+				} }
+			>
+				<ExpressCheckoutElement
+					options={ {
+						paymentMethods: {
+							amazonPay: 'never',
+							applePay:
+								paymentMethod === 'applePay'
+									? 'always'
+									: 'never',
+							googlePay:
+								paymentMethod === 'googlePay'
+									? 'always'
+									: 'never',
+							link: 'never',
+							paypal: 'never',
+						},
+					} }
+					onReady={ ( event ) => {
+						let canMakePayment = false;
+						if ( event.availablePaymentMethods ) {
+							canMakePayment =
+								event.availablePaymentMethods[ paymentMethod ];
+						}
+						// eslint-disable-next-line no-console
+						console.log(
+							`Payment method ${ paymentMethod }, is available? `,
+							canMakePayment
+						);
+						resolve( canMakePayment );
```

Comment:
Do you think we should do any clean up here and unmount this element once we've gotten what we need?

Reply:
Good call. I'll see what can we do.

Reply:
done [3247c13](https://github.com/Automattic/woocommerce-payments/pull/9378/commits/3247c137752402e65299ecfa21f5aa8083d200be)